File: unfuddle/whatbi/ports/WhatAGem/Framework/tests/map.rb
require "WhatAGem/Framework/Map"
require "WhatAGem/Framework/Tests/DataSource"

module WhatAGem
	module Tests
		class MapTest < DataSourceTestBase
			def dataSourceClass
				MapDataSource
			end
			def testInvalidGetNonExistent
				assert_raise(WhatAGem::InvalidKeyException) { @ds.get("Dino") }
			end
		end # MapTest
	end # Tests
end # WhatAGem

File: unfuddle/whatbi/Framework/trunk/Tests/Map.cs
/*
 *	Project:		WhaTBI? (What's The Big Idea?)
 *	Filename:		Tests/ReadOnlyMap.cs
 *	Description:	Test of simple, map-based datasource
 */

using System;
using System.Collections;

#if USE_MBUNIT
using MbUnit.Core.Framework;
using MbUnit.Framework;
#else
using NUnit.Framework;
#endif

using Pdbartlett.Whatbi;
using Pdbartlett.Whatbi.Examples;

namespace Pdbartlett.Whatbi.Tests
{
	[TestFixture] public class MapTest : DataSourceBaseTest
	{
		// initialisation/destruction

		protected override IDataSource CreateDataSource()
		{
			return new MapDataSource(TestData);
		}
	}
	
	[TestFixture] public class UpdatableMapTest : UpdatableDataSourceBaseTest 
	{		
		// initialisation/destruction

		protected override IDataSource CreateDataSource()
		{
			return new UpdatableMap(TestData);
		}
	}
}

